tensorboard/uploader/exporter.py

```diff
@@ -110,13 +110,19 @@ def export(self, read_time=None):
       read_time = time.time()
     for experiment_id in self._request_experiment_ids(read_time):
       filepath = _scalars_filepath(self._outdir, experiment_id)
-      with _open_excl(filepath) as outfile:
-        data = self._request_scalar_data(experiment_id, read_time)
-        for block in data:
-          json.dump(block, outfile, sort_keys=True)
-          outfile.write("\n")
-          outfile.flush()
-      yield experiment_id
+      try:
+        with _open_excl(filepath) as outfile:
+          data = self._request_scalar_data(experiment_id, read_time)
+          for block in data:
+            json.dump(block, outfile, sort_keys=True)
+            outfile.write("\n")
+            outfile.flush()
+        yield experiment_id
+      except grpc.RpcError as e:
+        if e.code() == grpc.StatusCode.CANCELLED:
+          raise GrpcTimeoutException(experiment_id)
+        else:
+          raise(e)
```

Comment:
Use `raise` (not `raise e`) to reraise without losing the original
traceback.

Reply:
Done.